cni/pkg/nodeagent/server.go

```diff
@@ -0,0 +1,270 @@
+	probeSet, err := ipset.NewIPPortSet(constants.ProbeIPSet, linDeps)
+	if err != nil {
+		return probeSet, err
+	}
+	probeSet.Flush()
+	return probeSet, nil
+}
+
+func (s *Server) Start() {
+	log.Info("CNI ambient server starting")
+	s.kubeClient.RunAndWait(s.ctx.Done())
+	log.Info("CNI ambient server kubeclient started")
+	pods := s.handlers.GetAmbientPods()
+	err := s.dataplane.ConstructInitialSnapshot(pods)
+	if err != nil {
+		log.Warnf("failed to construct initial snapshot: %v", err)
+	}
+
+	log.Info("CNI ambient server marking ready")
+	s.Ready()
+	s.dataplane.Start(s.ctx)
+	s.handlers.Start()
+}
+
+func (s *Server) Stop() {
+	log.Info("CNI ambient server terminating, cleaning up node net rules")
+
+	s.cniServerStopFunc()
+	s.dataplane.Stop()
+}
+
+type meshDataplane struct {
+	kubeClient   kubernetes.Interface
+	netServer    MeshDataplane
+	redirectMode RedirectMode
+}
+
+func (s *meshDataplane) Start(ctx context.Context) {
+	s.netServer.Start(ctx)
+}
+
+func (s *meshDataplane) Stop() {
+	s.netServer.Stop()
+}
+
+func (s *meshDataplane) ConstructInitialSnapshot(ambientPods []*corev1.Pod) error {
+	switch s.redirectMode {
+	case InPodMode:
+		return s.netServer.ConstructInitialSnapshot(ambientPods)
+
+	default:
+		log.Errorf("unknown redirect mode: %v", s.redirectMode)
+		return fmt.Errorf("unknown redirect mode: %v", s.redirectMode)
+	}
+}
+
+func (s *meshDataplane) AddPodToMesh(ctx context.Context, pod *corev1.Pod, podIPs []netip.Addr, netNs string) error {
+	var retErr error
+	switch s.redirectMode {
+	case InPodMode:
+		err := s.netServer.AddPodToMesh(ctx, pod, podIPs, netNs)
+		if err != nil {
+			log.Errorf("failed to add pod to ztunnel: %v", err)
+			if !errors.Is(err, ErrPartialAdd) {
+				return err
+			}
+			retErr = err
+		}
+
+	default:
+		log.Errorf("unknown redirect mode: %v", s.redirectMode)
+		return fmt.Errorf("unknown redirect mode: %v", s.redirectMode)
+	}
+
+	log.Debugf("annotating pod %s", pod.Name)
+	if err := util.AnnotateEnrolledPod(s.kubeClient, &pod.ObjectMeta); err != nil {
+		log.Errorf("failed to annotate pod enrollment: %v", err)
+		// don't return error here, as this is purely informational.
+	}
+	return retErr
+}
+
+func (s *meshDataplane) RemovePodFromMesh(ctx context.Context, pod *corev1.Pod) error {
+	log := log.WithLabels("ns", pod.Namespace, "name", pod.Name)
+	log.Debug("Pod is now stopped or opt out... cleaning up.")
+	switch s.redirectMode {
+	case InPodMode:
+		err := s.netServer.RemovePodFromMesh(ctx, pod)
+		if err != nil {
+			log.Errorf("failed to remove pod to ztunnel: %v", err)
+			return err
+		}
+	default:
+		log.Errorf("unknown redirect mode: %v", s.redirectMode)
+	}
+	log.Debug("removing annotation from pod")
+	err := util.AnnotateUnenrollPod(s.kubeClient, &pod.ObjectMeta)
+	if err != nil {
+		log.Errorf("failed to annotate pod unenrollment: %v", err)
+	}
+	return err
+}
+
+// Delete pod from mesh: pod is deleted. iptables rules will die with it, we just need to update zutnnel
```

Comment:
```suggestion
// Delete pod from mesh: pod is deleted. iptables rules will die with it, we just need to update ztunnel
```
nit: typo of ztunnel